### PR TITLE
Score

### PR DESCRIPTION
**Pull request by Ceanze**
- There now is a score manager which counts score.
- Replay phase now shows with GUI the score and targets hit.

Keep in mind the score counting is not finalized, but the main principle is there.

### CODE REVIEW

File: GameProject/Game/GameLogic/ReplayPhase.cpp [hunk 1]
```diff
@@ -11,6 +11,14 @@
 #include <Game/GameLogic/GuidingPhase.h>
 #include <Game/GameLogic/AimPhase.h>
 
+// Gui
+#include <Engine/GUI/FontManager.h>
+#include <Engine/GUI/Panel.h>
+#include <Engine/GUI/Button.h>
+
+// Remove later
+#include <Utils/Logger.h>
```
Comment (TheoBerlin) [inline GameProject/Game/GameLogic/ReplayPhase.cpp]: Remove

File: GameProject/Game/GameLogic/ReplayPhase.cpp [hunk 2]
```diff
@@ -11,6 +11,14 @@
 #include <Game/GameLogic/GuidingPhase.h>
 #include <Game/GameLogic/AimPhase.h>
 
+// Gui
```
Comment (TheoBerlin) [inline GameProject/Game/GameLogic/ReplayPhase.cpp]: I don't think we have a need for this kind of categorization.

File: GameProject/Game/GameLogic/ReplayPhase.h [hunk 3]
```diff
@@ -25,13 +28,24 @@ class ReplayPhase : public Phase
 
     void transitionToAim(CameraTransitionEvent* event);
 
+	void activateGui();
```
Comment (TheoBerlin) [inline GameProject/Game/GameLogic/ReplayPhase.h]: This name is very vague and will be mistaken for something else once more GUI is implemented into `ReplayPhase`. How about `showResults` or `presentScore`?

File: GameProject/Game/GameLogic/ReplayPhase.h [hunk 4]
```diff
@@ -25,13 +28,24 @@ class ReplayPhase : public Phase
...
+	void guiCallback();
+
     Entity* freeCam;
     FreeMove* freeMove;
 
     Entity* replayArrow;
     PathTreader* pathTreader;
     PathVisualizer* pathVisualizer;
 
+	// Gui stuff
```
Comment (TheoBerlin) [inline GameProject/Game/GameLogic/ReplayPhase.h]: Again, avoid the vague word of GUI.

File: GameProject/Game/GameLogic/ScoreManager.cpp [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+	this->totalTime = 0.0f;
+	this->timer->restart();
+}
+
+void ScoreManager::stop()
+{
+	this->timer->stop();
+	this->totalTime = this->timer->getTime();
+	this->counting = false;
+	// Calculate the new total score with time as a factor
```
Comment (TheoBerlin) [inline GameProject/Game/GameLogic/ScoreManager.cpp]: Break this up with an empty line.

File: GameProject/Game/GameLogic/ReplayPhase.cpp (hunk 1 above)
Comment (Ceanze) [reply GameProject/Game/GameLogic/ReplayPhase.cpp]: Removed

File: GameProject/Game/GameLogic/ReplayPhase.cpp (hunk 2 above)
Comment (Ceanze) [reply GameProject/Game/GameLogic/ReplayPhase.cpp]: Removed

File: GameProject/Game/GameLogic/ReplayPhase.h (hunk 3 above)
Comment (Ceanze) [reply GameProject/Game/GameLogic/ReplayPhase.h]: It was indeed a bad name, could not figure out a better one, went with showResults now.

File: GameProject/Game/GameLogic/ReplayPhase.h (hunk 4 above)
Comment (Ceanze) [reply GameProject/Game/GameLogic/ReplayPhase.h]: Renamed

File: GameProject/Game/GameLogic/ScoreManager.cpp (hunk 5 above)
Comment (Ceanze) [reply GameProject/Game/GameLogic/ScoreManager.cpp]: \n

File: GameProject/Game/GameLogic/TargetManager.cpp [hunk 6]
```diff
@@ -68,6 +68,11 @@ void TargetManager::resetTargets()
     resetStaticTargets();
 }
 
+unsigned TargetManager::getTotalTargets()
```
Comment (TheoBerlin) [inline GameProject/Game/GameLogic/TargetManager.cpp]:
```suggestion
unsigned TargetManager::getTargetCount()
```

File: GameProject/Game/GameLogic/ReplayPhase.h [hunk 7]
```diff
@@ -25,13 +28,24 @@ class ReplayPhase : public Phase
...
     PathTreader* pathTreader;
     PathVisualizer* pathVisualizer;
 
+	// Results panel varibles
+	float timeLeft;
+	bool showGui;
+	bool minimized;
+	Panel* smallPanel;
+	Panel* bigPanel;
+
```
Comment (TheoBerlin) [inline GameProject/Game/GameLogic/ReplayPhase.h]: Unnecessary empty line

Comment (Ceanze) [reply GameProject/Game/GameLogic/ReplayPhase.h]: No longer relevant

File: GameProject/Game/GameLogic/TargetManager.cpp (hunk 6 above)
Comment (Ceanze) [reply GameProject/Game/GameLogic/TargetManager.cpp]: Renamed

**Pull request merged**